instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/cache/Cache.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.cache;
+
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+/** A cache from keys to values.
+ *
+ * Keys are always referenced weakly and are compared using identity comparison, not
+ * {@link Object#equals(Object)}.
+ */
+public interface Cache<K, V> {
```

Comment:
Do we want to keep the name `Cache`? It's now just a weak-keyed map, since we removed caffeine et al.

Reply:
I did not want to rename in this PR, to avoid too many changes at once. We can rename later, if we want

Reply:
Now there are two different implementations, so we better keep this generic name